orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/StartStageHandler.kt

```diff
@@ -77,12 +77,14 @@ class StartStageHandler(
           queue.push(SkipStage(stage))
         } else {
           try {
+            // Set the startTime in case we throw an exception.
+            stage.startTime = clock.millis()
+            repository.storeStage(stage)
+
             stage.withAuth {
               stage.plan()
             }
-
             stage.status = RUNNING
```

Comment:
I think it's actually better to have this together with `stage.startTime` as @robfletcher noted. Looking at `CompleteStageHandler` (https://github.com/spinnaker/orca/blob/master/orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandler.kt#L56) it seems that it will be harmless. And it will be more accurate